image-builder/image_builder.sh

```diff
@@ -193,12 +194,18 @@ check_rootfs() {
 			;;
 
 		"yes")
-			agent_path="/usr/bin/${AGENT_BIN}"
+			agent_path="/sbin/init"
 			agent="${rootfs}${agent_path}"
 			if  [ ! -x "${agent}" ]; then
 				error "${agent_path} is not installed in ${rootfs}. Use AGENT_BIN env variable to change the expected agent binary name"
 				return 1
 			fi
+			# checksum must be different to system
+			if [ -f "${systemd}" ] && [ "$(cmp -s "${systemd}" "${agent}")" ]; then
```

Comment:
That `cmp` doesn't look right to me - you're checking the *output* of `cmp`, but running with `-s` meaning "no output"... ? Can't it be:

```sh
if [ -f "${systemd}" ] && cmp -s "${systemd}" "${agent}"; then
```

Reply:
good catch! you're right, thanks